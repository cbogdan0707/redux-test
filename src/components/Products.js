import React from "react";
import { connect } from "react-redux";

import { onProductClick } from "../actions";

const displayProducts = (productsList, activeProducts, onProductClick) => {
  return productsList.map(product => {
    const activeClass = `item ${
      activeProducts.includes(product.id) ? "active" : ""
    }`;
    return (
      <div
        className={activeClass}
        key={product.id}
        onClick={() => onProductClick(product.id)}
      >
        {product.title}
        <div>{activeProducts.includes(product.id) && product.description}</div>
      </div>
    );
  });
};

export const Products = ({
  activeProducts,
  onProductClick,
  productsByCategory,
  products,
  emptyResults,
  inputFilterProducts
}) => {
  if (emptyResults === 0) {
    return null;
  }
  return inputFilterProducts && inputFilterProducts.length
    ? displayProducts(inputFilterProducts, activeProducts, onProductClick)
    : productsByCategory && productsByCategory.length
    ? displayProducts(productsByCategory, activeProducts, onProductClick)
    : displayProducts(products, activeProducts, onProductClick);
};

const mapStateToProps = state => {
  return {
    activeProducts: state.activeProducts,
    productsByCategory: state.productsByCategory.filteredProducts,
    products: state.products,
    emptyResults: state.productsByCategory.emptyResults,
    inputFilterProducts: state.productsByCategory.inputFilteredProducts
  };
};

export default connect(
  mapStateToProps,
  { onProductClick }
)(Products);
