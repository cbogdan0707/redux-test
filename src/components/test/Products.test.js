import React from "react";
import renderer from "react-test-renderer";

import { Products } from "../Products";
import productsMock from "../../mocks/productsMock";

describe("Products snapshot test", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Products productsByCategory={productsMock.data} activeProducts="" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
