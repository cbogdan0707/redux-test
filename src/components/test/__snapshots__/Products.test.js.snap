// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products snapshot test renders correctly 1`] = `
Array [
  <div
    className="item "
    onClick={[Function]}
  >
    Borsao Macabeo
    <div />
  </div>,
  <div
    className="item "
    onClick={[Function]}
  >
    Domaine de L'Olibet 'Les Pujols' Cinsault Ros√©
    <div />
  </div>,
  <div
    className="item "
    onClick={[Function]}
  >
    Choc on Choc Make Your Own Egg Kit
    <div />
  </div>,
]
`;
